frame/support/test/tests/construct_runtime.rs

```diff
@@ -948,40 +948,66 @@ fn test_metadata() {
 fn pallet_in_runtime_is_correct() {
 	assert_eq!(PalletInfo::index::<System>().unwrap(), 30);
 	assert_eq!(PalletInfo::name::<System>().unwrap(), "System");
+	assert_eq!(PalletInfo::module_name::<System>().unwrap(), "system");
+	assert_eq!(PalletInfo::crate_version::<System>().unwrap(), CrateVersion::new(3, 0, 0));
 
 	assert_eq!(PalletInfo::index::<Module1_1>().unwrap(), 31);
 	assert_eq!(PalletInfo::name::<Module1_1>().unwrap(), "Module1_1");
+	assert_eq!(PalletInfo::module_name::<Module1_1>().unwrap(), "module1");
+	assert_eq!(PalletInfo::crate_version::<Module1_1>().unwrap(), CrateVersion::new(3, 0, 0));
 
 	assert_eq!(PalletInfo::index::<Module2>().unwrap(), 32);
 	assert_eq!(PalletInfo::name::<Module2>().unwrap(), "Module2");
+	assert_eq!(PalletInfo::module_name::<Module2>().unwrap(), "module2");
+	assert_eq!(PalletInfo::crate_version::<Module2>().unwrap(), CrateVersion::new(3, 0, 0));
 
 	assert_eq!(PalletInfo::index::<Module1_2>().unwrap(), 33);
 	assert_eq!(PalletInfo::name::<Module1_2>().unwrap(), "Module1_2");
+	assert_eq!(PalletInfo::module_name::<Module1_2>().unwrap(), "module1");
+	assert_eq!(PalletInfo::crate_version::<Module1_2>().unwrap(), CrateVersion::new(3, 0, 0));
 
 	assert_eq!(PalletInfo::index::<NestedModule3>().unwrap(), 34);
 	assert_eq!(PalletInfo::name::<NestedModule3>().unwrap(), "NestedModule3");
+	assert_eq!(PalletInfo::module_name::<NestedModule3>().unwrap(), "nested_module3");
+	assert_eq!(PalletInfo::crate_version::<NestedModule3>().unwrap(), CrateVersion::new(3, 0, 0));
 
 	assert_eq!(PalletInfo::index::<Module3>().unwrap(), 35);
 	assert_eq!(PalletInfo::name::<Module3>().unwrap(), "Module3");
+	assert_eq!(PalletInfo::module_name::<Module3>().unwrap(), "self_module3");
```

Comment:
This one here is a bit odd. I have half a mind to instruct the module_name method to ignore `self` and `crate` keywords when it sees them appearing at the beginning of the path...

Reply:
I'm not sure, what if the path is exaclty `crate` ?